reactor-core/src/main/java/reactor/core/publisher/DirectProcessor.java

```diff
@@ -77,7 +78,9 @@
  * </p>
  *
  * @param <T> the input and output value type
+ * @deprecated Prefer clear cut usage of either {@link Processors} or {@link Sinks}, to be removed in 3.5
```

Comment:
Can these wait until a major to be removed?

Reply:
can you expand on your concern ? if we follow the history of Reactor (2.x > 3.0 > 3.1 > 3.2 and now 3.3), a "major" would be a significant event that would probably mean a complete re-architecture, to the point that the whole ecosystem around Reactor would probably need heavy re-architecturing too, or break (see Reactor 2 vs Reactor 3). The only other drives for a Reactor 4 that I could think about are a change in the Java baseline or the advent of a Reactive Streams v2. In any case, very unlikely at this point.

Reply:
deprecating one release and removing the next just breaks backward compatibility quickly. With boot moving to a 6-month cadence for minors, they are deprecating, then waiting an extra release to remove. So deprecate in x.1, do nothing in x.2 and remove in x.3.

Reply:
I see. But unlike Boot, we haven't switched to a 6 month cadence. So deprecating these now would mean that the concrete classes would get hidden in a year or so. Initially, there was an issue to directly remove them without the deprecation, but I agree this would be overkill. Note that `TopicProcessor` and `WorkQueueProcessor` are slated to be removed though (since deprecated in 3.3.0)

Reply:
:+1: got it. thanks for taking the time to explain.